integration-test/src/test/java/org/ehcache/integration/statistics/JCacheCalculationTest.java

```diff
@@ -0,0 +1,372 @@
+package org.ehcache.integration.statistics;
+
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+import javax.cache.Cache;
+import javax.cache.CacheManager;
+import javax.cache.Caching;
+import javax.cache.spi.CachingProvider;
+
+import org.ehcache.config.CacheConfiguration;
+import org.ehcache.config.builders.CacheConfigurationBuilder;
+import org.ehcache.config.builders.ResourcePoolsBuilder;
+import org.ehcache.core.EhcacheManager;
+import org.ehcache.core.config.DefaultConfiguration;
+import org.ehcache.core.spi.service.StatisticsService;
+import org.ehcache.impl.config.persistence.DefaultPersistenceConfiguration;
+import org.ehcache.jsr107.EhcacheCachingProvider;
+import org.ehcache.spi.service.Service;
+import org.ehcache.spi.service.ServiceProvider;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Check that calculations are accurate according to specification. Each cache method have a different impact on the statistics
+ * so each method should be tested
+ */
+public class JCacheCalculationTest extends AbstractCacheCalculationTest {
```

Comment:
my IDE got nervous with all the "duplicates" from CalculationTest.
I'm ok with that personally, as long as we think that, it's possible that stats expectations could differ between jcache and ehcache in the future